cmd/queue/main.go

```diff
@@ -121,6 +117,7 @@ var (
 		stats.UnitDimensionless)
 
 	readinessProbeTimeout = flag.Int("probe-period", -1, "run readiness probe with given timeout")
+	readinessProbePort    = flag.Int("port", -1, "run readiness probe to the given port")
```

Comment:
Can we read this from the environment? `QUEUE_SERVING_PORT` should already contain this I think.

Reply:
Will this work given that the exec probe runs as a separate process?

Reply:
IIUC, it has to read `QUEUE_SERVING_PORT` first from env variables each time. So I prefer to pass it as flag.

Reply:
Yeah we'd read it from env in line 294. It'd just safe you a ton of changes in this PR since we already pass it to the container where this binary is supposed to run. If the env is identical (I haven't tried that) we could even parse it via the config helpers we have here.

Reply:
Well, the changes have already been made (some changes are good refactoring) and I still prefer not to read env variable every time.

Reply:
If we already passed this as env var then we shouldn't pass another flag.  Reading from the same mechanism will ensure consistency.

Reply:
If we are ok to read a bunch of environment variables every time when calling this binary, I can use it instead of introducing a new flag.

Reply:
Yeah, I think that's fine.  Reading env var once per run isn't that expensive, especially when we put that next to the cost of launching a binary and sending HTTP requests.  If we do this in a loop I'd worry more.

May be best to start with reusing the env var and optimize if that becomes a problem?

Reply:
I have tested it. Reading env variables costs < 0.5 ms. I have removed the port flag. PTAL.